xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -194,14 +201,14 @@ public void reset() {
     public void updateLocalityStore(Map<XdsLocality, LocalityInfo> localityInfoMap) {
       Set<XdsLocality> oldLocalities = localityMap.keySet();
       Set<XdsLocality> newLocalities = localityInfoMap.keySet();
-      Map<XdsLocality, LocalityLbInfo> updatedLocalityMap = new LinkedHashMap<>();
+      ImmutableMap.Builder<XdsLocality, LocalityLbInfo> updatedLocalityMap = ImmutableMap.builder();
 
       final Set<XdsLocality> toRemove = new HashSet<>();
       for (XdsLocality oldLocality : oldLocalities) {
-        if (!newLocalities.contains(oldLocality)) {
+        if (!newLocalities.contains(oldLocality)
+            && !localityMap.get(oldLocality).isDeactivated()) {
+          deactivate(oldLocality);
           toRemove.add(oldLocality);
```

Comment:
`toRemove` is actually `toBeRemovedFromStatsStore`. Should name it more clearly.

Reply:
Done.